git/githistory/log/log.go

```diff
@@ -0,0 +1,197 @@
+package log
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"strings"
+	"sync"
+
+	"github.com/git-lfs/git-lfs/tools"
+	"github.com/olekukonko/ts"
+)
+
+// Logger logs a series of tasks to an io.Writer, processing each task in order
+// until completion .
+type Logger struct {
+	// sink is the writer to write to.
+	sink io.Writer
+
+	// widthFn is a function that returns the width of the terminal that
+	// this logger is running within.
+	widthFn func() int
+
+	// queue is the incoming, unbuffered queue of tasks to enqueue.
+	queue chan Task
+	// tasks is the set of tasks to process.
+	tasks chan Task
```

Comment:
If you skip `queue` and just use `tasks`, it should remove a lot of of complexity from `consume()`. The comment sounds like you intended for `tasks` to be buffered, but it's not.